src/main/java/com/redislabs/redistimeseries/CreateParams.java

```diff
@@ -0,0 +1,113 @@
+package com.redislabs.redistimeseries;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import redis.clients.jedis.Protocol;
+import redis.clients.jedis.util.SafeEncoder;
+
+public class CreateParams {
+
+  private Long retentionTime;
+  private boolean uncompressed;
+  private Long chunkSize;
+  private DuplicatePolicy duplicatePolicy;
+  private Map<String, String> labels;
+
+  public static CreateParams createParams() {
+    return new CreateParams();
+  }
+
+  public CreateParams() {}
+
+  public CreateParams retentionTime(long retentionTime) {
+    this.retentionTime = retentionTime;
+    return this;
+  }
+
+  public CreateParams uncompressed() {
+    this.uncompressed = true;
+    return this;
+  }
+
+  public CreateParams chunkSize(long chunkSize) {
+    this.chunkSize = chunkSize;
+    return this;
+  }
+
+  public CreateParams duplicatePolicy(DuplicatePolicy duplicatePolicy) {
+    this.duplicatePolicy = duplicatePolicy;
+    return this;
+  }
+
+  public CreateParams label(String label, String value) {
+    if (this.labels == null) this.labels = new HashMap<>();
+    this.labels.put(label, value);
+    return this;
+  }
+
+  public CreateParams labels(String... labels) {
```

Comment:
@sazzad16 I think it worth a JDoc to explain how you "parse" the labels

Reply:
@gkorland I added this method in my previous approach where I was calling new `add` method from all existing `add` methods. This is not required in this approach.